application/controllers/admin/ConditionsAction.php

```diff
@@ -1484,9 +1484,10 @@ protected function getCAnswersAndCQuestions(array $theserows)
                             'questionl10ns' => array(
                                 'condition' => 'questionl10ns.language = :lang',
                                 'params' => array(':lang' => $this->language)
-                            )))->findAllByAttributes(array('parent_qid' => $rows['qid']), array('order' => 'question_order desc'));
+                            )))->findAllByAttributes(array('parent_qid' => $rows['qid']), array('order' => 'question_order ASC, scale_id ASC'));
                 foreach ($aresult as $arows) {
-                    $attr = QuestionAttribute::model()->getQuestionAttributes($rows['qid']);
+                    /* Used to get dualscale_headerA and dualscale_headerB */
+                    $attr = QuestionAttribute::model()->getQuestionAttributes(Question::model()->findByPk($rows['qid']));
```

Comment:
Instead of loading a question object individually for all rows that have subquestions we should pass the questions as part of `$theserows` to the method, loaded from the database with a single query, so if we end up having many questions, we send a single query to the database instead of sending individually a query for each row (we do not send multiple individual queries for each row even if they have multiple matches because there is a caching mechanism after each individual query)

Reply:
Argl … ! Thank you !